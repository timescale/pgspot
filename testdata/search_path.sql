
-- should warn since its unqualified function call
SELECT unsafe_call3('%s','abc');
SELECT pg_catalog.safe_call4();

SET search_path TO pg_catalog, pg_temp;

-- safe since we have safe search_path now
SELECT safe_call9('%s','abc');
SELECT pg_catalog.safe_call10();

RESET search_path;

-- should warn since search_path got reset
SELECT unsafe_call15('%s','abc');
SELECT pg_catalog.safe_call16();

CREATE SCHEMA new;

SELECT new.safe_call20('%s','abc');
SELECT unsafe_call21('%s','abc');

