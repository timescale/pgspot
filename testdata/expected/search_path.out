PS016: Unqualified function call: unsafe_call3 at line 2
PS016: Unqualified function call: unsafe_call15 at line 13
PS016: Unqualified function call: unsafe_call21 at line 21

Errors: 0 Warnings: 3 Unknown: 0
